packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts

```diff
@@ -916,6 +917,32 @@ export function extractPageId(page = '') {
 	return page;
 }
 
+export function getPageId(this: IExecuteFunctions, i: number) {
+	const page = this.getNodeParameter('pageId', i, {}) as INodeParameterResourceLocator;
+	let pageId = '';
+
+	if (page.value && typeof page.value === 'string') {
+		if (page.mode === 'id') {
+			pageId = page.value;
+		} else if (page.value.includes('p=')) {
+			// e.g https://www.notion.so/xxxxx?v=xxxxx&p=xxxxx&pm=s
+			pageId = page.value.split('p=')[1].split('&')[0];
+		} else {
+			// e.g https://www.notion.so/page_name-xxxxx
+			pageId = page.value.match(databasePageUrlValidationRegexp)?.[1] || '';
```

Comment:
nit: `??`

Reply:
this how it works in rlc when extracting value, taking match from regex